moxin-frontend/src/data/download.rs

```diff
@@ -66,7 +66,7 @@ impl Download {
         });
     }
 
-    pub fn update_download_progress(&mut self) {
+    pub fn process_download_progress(&mut self) {
```

Comment:
Minor (and very opinionated) renaming - I'd argue it's job is to process progress updates, and the backend is the one updating the progress